docs/source/platform/federation.mdx

```diff
@@ -0,0 +1,354 @@
+                      --serviceURL="http://launches-graphql.svc.cluster.local:4001/" \
+                      --endpoint="http://localhost:4001/"
+  ✔ Loading Apollo Project
+  ✔ Uploading service to Engine
+
+The 'registry' service for the 'space-explorer@current' graph was updated
+
+The gateway for the 'space-explorer@current' graph was updated with a new schema, composed from the updated 'launches' service
+
+id      graph             variant
+──────  ────────────────  ───────
+az329e  space-explorer    current
+```
+
+> What's the difference between `serviceURL` and `endpoint` parameters? The `endpoint` parameter controls the endpoint where the schema will be fetched from at composition, whereas `serviceURL` controls what URL the gateway will query at runtime. This is especially helpful if you have disabled introspection queries in your production services.
+
+It's important to make sure that any possible end-user effect from the changes to the graph have been identified, and it's similarly important to strive for backwards-compatible changes to limit those effects. The reason for waiting for the service to completely roll over before registering it is that if some services are still exposing the previous configuration, they might elicit failures for operations the gateway has planned with the new configuration.
+
+### Diving into service:push
+
+Every time that `apollo service:push` is called for a federated service, it not only registers the implementing service to the graph, but it also updates the managed configuration files that the gateway has access to. Because the graph is dynamically changing, it's possible for composition errors to occur for a `service:push` even after a `service:check` has succeeded if other implementing services changed in the interim. For this reason, updating an implementing service will re-trigger composition in the cloud, ensuring that the implementing services still compose to form a complete graph before provisioning the managed configuration. The workflow behind the scenes can be summed up as follows:
+
+1. The partial schema is uploaded to a secure location and indexed
+1. The implementing service is updated in the registry to point to the partial schema
+1. All implementing services are composed in the cloud to produce a new complete schema
+1. If composition fails, the command exits and emits errors
+1. If composition succeeds, the top-level managed configuration file is updated in-place to point to the updated set of implementing services
+
+On the other side of the equation, sits the gateway. The gateway is constantly listening for changes to the top-level managed configuration file. The location of the managed configuration file is guarded by using a hash of the API key, provisioned ahead of time so as not to affect reliability. The life-cycle of dynamic configuration updates is as follows:
+
+1. The gateway listens for updates to its managed configuration every 10 seconds
```

Comment:
Every 10s is an implementation detail that will be configurable and possibly changed down the road, I'll suggest staying slightly less specific here.
```suggestion
1. The gateway listens for updates to its managed configuration
```